aws/resource_aws_qldb_ledger_test.go

```diff
@@ -0,0 +1,249 @@
+package aws
+
+import (
+	"fmt"
+	"log"
+	"regexp"
+	"testing"
+	"time"
+
+	"github.com/aws/aws-sdk-go/service/qldb"
+	"github.com/hashicorp/terraform-plugin-sdk/helper/acctest"
+	"github.com/hashicorp/terraform-plugin-sdk/helper/resource"
+	"github.com/hashicorp/terraform-plugin-sdk/helper/schema"
+	"github.com/hashicorp/terraform-plugin-sdk/terraform"
+
+	"github.com/aws/aws-sdk-go/aws"
+)
+
+func init() {
+	resource.AddTestSweepers("aws_qldb_ledger", &resource.Sweeper{
+		Name: "aws_qldb_ledger",
+		F:    testSweepQLDBLedgers,
+	})
+}
+
+func testSweepQLDBLedgers(region string) error {
+	client, err := sharedClientForRegion(region)
+
+	if err != nil {
+		return fmt.Errorf("error getting client: %s", err)
+	}
+
+	conn := client.(*AWSClient).qldbconn
+	input := &qldb.ListLedgersInput{}
+	page, err := conn.ListLedgers(input)
+
+	if err != nil {
+		if testSweepSkipSweepError(err) {
+			log.Printf("[WARN] Skipping QLDB Ledger sweep for %s: %s", region, err)
+			return nil
+		}
+		return fmt.Errorf("Error listing QLDB Ledgers: %s", err)
+	}
+
+	for _, item := range page.Ledgers {
+		input := &qldb.DeleteLedgerInput{
+			Name: item.Name,
+		}
+		name := aws.StringValue(item.Name)
+
+		log.Printf("[INFO] Deleting QLDB Ledger: %s", name)
+		_, err = conn.DeleteLedger(input)
+
+		if err != nil {
+			log.Printf("[ERROR] Failed to delete QLDB Ledger %s: %s", name, err)
+			continue
+		}
+
+		if err := waitForQLDBLedgerDeletion(conn, name); err != nil {
+			log.Printf("[ERROR] Error waiting for QLDB Ledger (%s) deletion: %s", name, err)
+		}
+	}
+
+	return nil
+}
+
+func TestAccAWSQLDBLedger_basic(t *testing.T) {
+	var qldbCluster qldb.DescribeLedgerOutput
+	rInt := acctest.RandInt()
+	resourceName := "aws_qldb_ledger.test"
+
+	resource.ParallelTest(t, resource.TestCase{
+		PreCheck:     func() { testAccPreCheck(t) },
+		Providers:    testAccProviders,
+		CheckDestroy: testAccCheckAWSQLDBLedgerDestroy,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccAWSQLDBLedgerConfig(rInt),
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckAWSQLDBLedgerExists(resourceName, &qldbCluster),
+					resource.TestMatchResourceAttr(resourceName, "arn", regexp.MustCompile(`^arn:[^:]+:qldb:[^:]+:\d{12}:ledger/.+`)),
+					resource.TestMatchResourceAttr(resourceName, "name", regexp.MustCompile("test-ledger-[0-9]+")),
+					resource.TestCheckResourceAttr(resourceName, "permissions_mode", "ALLOW_ALL"),
+					resource.TestCheckResourceAttr(resourceName, "deletion_protection", "false"),
+					resource.TestCheckResourceAttr(resourceName, "tags.%", "0"),
+				),
+			},
+			{
+				ResourceName:      resourceName,
+				ImportState:       true,
+				ImportStateVerify: true,
+			},
+		},
+	})
+}
+
+func testAccCheckAWSQLDBLedgerDestroy(s *terraform.State) error {
```

Comment:
This function's definition should just be the logic of `testAccCheckAWSLedgerDestroyWithProvider` with `provider` replaced with `testAccProvider`